prime-router/src/test/kotlin/azure/WorkflowEngineTests.kt

```diff
@@ -160,6 +161,34 @@ class WorkflowEngineTests {
         confirmVerified(accessSpy, blobMock, queueMock) // todo
     }
 
+    @Test
+    fun `test verifyNoDuplicateFile`() {
+        mockkObject(BlobAccess.Companion)
+
+        val one = Schema(name = "one", topic = "test", elements = listOf(Element("a"), Element("b")))
+        val metadata = Metadata(schema = one)
+        val settings = FileSettings()
+        val sender = Sender("senderName", "org", Sender.Format.CSV, "covid-19", CustomerStatus.INACTIVE, one.name)
+
+        every {
+            accessSpy.checkForDuplicate(any(), any(), any())
+        }.returns(true)
+
+        val digest = "fakeDigest".toByteArray()
+        val engine = makeEngine(metadata, settings)
+        var ex: ActionError? = null
+        try {
+            engine.verifyNoDuplicateFile(sender, digest)
+        } catch (e: ActionError) {
+            ex = e
+        }
```

Comment:
can use `assertFailsWith` or it looks like the assertk library has a way of doing it as well. You can then follow up with more assertions about the exception

Reply:
Updated